rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -628,10 +630,22 @@ private void processSigning(long executionBlockNumber, BtcECKey federatorPublicK
                     inputScript.getChunks());
 
             if (!alreadySignedByThisFederator) {
-                int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorPublicKey);
-                inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSig.encodeToBitcoin(), sigIndex, 1, 1);
-                txIn.setScriptSig(inputScript);
-                logger.debug("Tx input for tx {} signed.", new Sha3Hash(rskTxHash));
+                try {
+                    int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorPublicKey);
+                    inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSig.encodeToBitcoin(), sigIndex, 1, 1);
+                    txIn.setScriptSig(inputScript);
+                    logger.debug("Tx input for tx {} signed.", new Sha3Hash(rskTxHash));
+                } catch (IllegalStateException e) {
+                    Federation retiringFederation = getRetiringFederation();
+                    if (getActiveFederation().getPublicKeys().contains(federatorPublicKey)) {
```

Comment:
According to what we've been looking at with @tinchou, you should probably be using the safer `Federation::hasPublicKey` instead. Same below with the retiring federation.